src/app/containers/Header/ScriptLink/index.test.jsx

```diff
@@ -0,0 +1,195 @@
+import React from 'react';
+import { node } from 'prop-types';
+import { MemoryRouter, Route } from 'react-router-dom';
+import { render, fireEvent } from '@testing-library/react';
+import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
+import '@testing-library/jest-dom/extend-expect';
+import { RequestContext } from '#contexts/RequestContext';
+import { ServiceContext } from '#contexts/ServiceContext';
+import { UserContext } from '#contexts/UserContext';
+import { service as serbianServiceConfig } from '#lib/config/services/serbian';
+import { service as ukChinaServiceConfig } from '#lib/config/services/ukchina';
+import * as cookies from '#contexts/UserContext/cookies';
+import ScriptLinkContainer, { getVariantHref } from '.';
+import {
+  articlePath,
+  cpsAssetPagePath,
+  errorPagePath,
+  frontPagePath,
+  legacyAssetPagePath,
+} from '#app/routes/utils/regex';
+
+const setPreferredVariantCookieSpy = jest.spyOn(
+  cookies,
+  'setPreferredVariantCookie',
+);
+
+const userContextMock = {
+  setPreferredVariantCookie: cookies.setPreferredVariantCookie,
+};
+
+const requestContextMock = {
+  variant: 'lat',
+};
+
+const withRouter = (component, matchPath, path) => {
+  const Wrapper = ({ children }) => (
+    <MemoryRouter initialEntries={[path]}>
+      <Route path={matchPath}>{children}</Route>
+    </MemoryRouter>
+  );
+
+  Wrapper.propTypes = {
+    children: node.isRequired,
+  };
+
+  return {
+    ...render(component, { wrapper: Wrapper }),
+  };
+};
+
+/* eslint-disable react/prop-types */
+const ScriptLinkContainerWithContext = ({
+  serviceContext = serbianServiceConfig.lat,
+  requestContext = requestContextMock,
+}) => (
+  <ServiceContext.Provider value={serviceContext}>
+    <UserContext.Provider value={userContextMock}>
+      <RequestContext.Provider value={requestContext}>
+        <ScriptLinkContainer />
+      </RequestContext.Provider>
+    </UserContext.Provider>
+  </ServiceContext.Provider>
+);
+
+describe(`Script Link`, () => {
+  shouldMatchSnapshot(
+    'should render correctly',
+    <MemoryRouter initialEntries={['/serbian/lat']}>
+      <Route path={frontPagePath}>
+        <ScriptLinkContainerWithContext />
+      </Route>
+    </MemoryRouter>,
+  );
+
+  describe('assertions', () => {
+    const testCases = {
+      article: {
+        matchPath: articlePath,
+        path: '/serbian/articles/c805k05kr73o/lat',
+        variantPath: '/serbian/articles/c805k05kr73o/cyr',
+      },
+      cpsAssetPage: {
+        matchPath: cpsAssetPagePath,
+        path: '/serbian/srbija-46748932/lat',
```

Comment:
The CPS Asset and Legacy Asset test cases should fail when this PR https://github.com/bbc/simorgh/pull/5718 is merged